BuildingBlocks/src/BuildingBlocks.Application/FluentValidation/ValueInValidator.cs

```diff
@@ -59,11 +62,28 @@ protected override string GetDefaultMessageTemplate(string errorCode)
 
 public static class ValidatorExtensions
 {
+    private static readonly ConcurrentDictionary<Type, MethodInfo> IS_VALID_CACHE = new();
+
     public static IRuleBuilderOptions<T, TProperty> In<T, TProperty>(this IRuleBuilder<T, TProperty> ruleBuilder,
         params TProperty[] validOptions)
     {
         return ruleBuilder
             .SetValidator(new ValueInValidator<T, TProperty>(validOptions))
             .WithErrorCode(GenericApplicationErrors.Validation.InvalidPropertyValue().Code);
     }
+
+    public static IRuleBuilderOptions<T, string?> ValidId<T, TId>(this IRuleBuilder<T, string?> ruleBuilder) where TId : StronglyTypedId
+    {
+        var method = IS_VALID_CACHE.FirstOrDefault(x => x.Key == typeof(TId)).Value;
+        if (method == null)
+        {
+            method = typeof(TId).GetMethod("IsValid")!;
```

Comment:
You shouldn't suppress null. Instead, make a null check and throw a descriptive error message, informing the developer about the missing method.

Reply:
Do you still want this if we are going to add an `ArchUnit` test that ensures all `StronglyTypedIds` have an `IsValid` method?

Reply:
I'd say: why not?

Reply:
Roger that.